buy_together_app/views.py

```diff
@@ -7,3 +9,34 @@ def main_page(request):
 
 def description_page(request):
     return render(request, 'buy_together_app/description.html')
+
+
+def signup_user(request, msg=""):
+    form = SupplierForm()
+    formm = ClientForm()
+    context = {'form': form, 'formm': formm, 'message': [msg]}
+    return render(request, 'buy_together_app/sign_up.html', context)
+
+
+def signup_supplier(request):
+    if request.method == 'POST':
+        form = SupplierForm(request.POST)
+        if form.is_valid():
+            try:
+                form.save()
+                return redirect('Main Page')
+            except Exception:
```

Comment:
Can we narrow down the exception? How can we tell that any exception will mean that this user is taken?

Reply:
So make the except statement only for that specific exception that is raised at this situation.

Reply:
Done!